sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -39,137 +64,178 @@ public sealed class ResourceIdentifier : IEquatable<ResourceIdentifier>, ICompar
-        {
+                if (!remaining.StartsWith(SubscriptionStart.AsSpan()) && !remaining.StartsWith(ProviderStart.AsSpan()))
+                    throw new ArgumentOutOfRangeException(nameof(resourceId), $"The ResourceIdentifier must start with {SubscriptionStart} or {ProviderStart}.");
+
+                remaining = remaining.Slice(1);
+                //trim trailing '/' off the end if it exists
+                if (remaining[remaining.Length - 1] == Separator)
+                    remaining = remaining.Slice(0, remaining.Length - 1);
+
+                ReadOnlySpan<char> nextWord = PopNextWord(ref remaining);
+
+                //we know we at least have 1 ResourceIdentifier in the tree here
+                ResourceIdentifierParts nextParts = GetNextParts(Root, ref remaining, ref nextWord);
+                while (!nextWord.IsEmpty)
+                {
+                    //continue to get the next ResourceIdentifier in the tree until we reach the end which will be 'this'
+                    ResourceIdentifier newParent = new ResourceIdentifier(nextParts.Parent, nextParts.ResourceType, nextParts.ResourceName, nextParts.IsProviderResource, nextParts.SpecialType);
+                    nextParts = GetNextParts(newParent, ref remaining, ref nextWord);
+                }
+
+                //initialize ourselves last
+                Init(nextParts.Parent, nextParts.ResourceType, nextParts.ResourceName, nextParts.IsProviderResource, nextParts.SpecialType);
+                _stringValue = resourceId;
+            }
         }
 
-        private ResourceIdentifier(ResourceIdentifier parent, string providerNamespace, string resourceTypeName, string resourceName)
-            : this(parent, new ResourceType(providerNamespace, resourceTypeName), resourceName, true)
+        private ResourceIdentifier(ResourceIdentifier? parent, ResourceType resourceType, string resourceName, bool isProviderResource, SpecialType? specialType)
         {
+            Init(parent, resourceType, resourceName, isProviderResource, specialType);
         }
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
 
-        private void Init(ResourceIdentifier parent, ResourceType resourceType, string resourceName, bool isProviderResource)
+        private void Init(ResourceIdentifier? parent, ResourceType resourceType, string resourceName, bool isProviderResource, SpecialType? specialType)
         {
             if (parent is not null)
             {
+                //if our parent is not null set the same common identifier values
                 Provider = parent.Provider;
                 SubscriptionId = parent.SubscriptionId;
                 Location = parent.Location;
                 ResourceGroupName = parent.ResourceGroupName;
             }
 
-            if (resourceType == SubscriptionResourceType)
+            if (specialType.HasValue)
```

Comment:
If the specialType enum has a 'None' value then this is no longer needed.

Reply:
resolving as duplicate